File: LeetCode_ProblemSet/Go/Problem1669.go
package main 

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func mergeInBetween(list1 *ListNode, a int, b int, list2 *ListNode) *ListNode {
	list2_head, list2_tail := list2, list2
	for list2_tail.Next != nil {
		list2_tail = list2_tail.Next
	}

	pre, post, ans := list1, list1, list1
	for i := 0; i < a - 1; i++ {
		pre = pre.Next
	}
	for i := 0; i < b + 1; i++ {
		post = post.Next
	}

	pre.Next = list2_head
	list2_tail.Next = post

	return ans
}

File: LeetCode_ProblemSet/Problem1669.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
    
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode list2_head = list2, list2_tail = list2;
        while (list2_tail.next != null) {
            list2_tail = list2_tail.next;
        }

        ListNode pre = list1, post = list1, ans = list1;
        for (int i = 0; i < a- 1; i++) {
            pre = pre.next;
        }
        for (int i = 0; i < b + 1; i++) {
            post = post.next;
        }

        pre.next = list2_head;
        list2_tail.next = post;

        return ans;
    }
}